src/gateway/dto/ventilation.py

```
@@ -26,30 +26,49 @@
 
 
 class VentilationDTO(BaseDTO):
-    def __init__(self, id, source=None, plugin=None, external_id='', name='', room=None, amount_of_levels=0,
-                 device_vendor='', device_type='', device_serial=''):
+    def __init__(self, id, source=None, plugin=None, external_id='', name='', room=None, amount_of_levels=1,
+                 min_level=0, max_level=0, device_vendor='', device_type='', device_serial=''):
         self.id = id  # type: int
         self.source = source  # type: SourceDTO
         self.plugin = plugin  # type: Optional[PluginDTO]
         self.external_id = external_id  # type: str
         self.name = name  # type: str
         self.room = room  # type: Optional[int]
-        self.amount_of_levels = amount_of_levels  # type: int
         self.device_vendor = device_vendor  # type: str
         self.device_type = device_type  # type: str
         self.device_serial = device_serial  # type: str
 
+        min_level, max_level, amount_of_levels = VentilationDTO.validate_levels(min_level, max_level, amount_of_levels)
+        self.amount_of_levels = amount_of_levels # type: int
+        self.min_level = min_level  # type: int
+        self.max_level = max_level  # type: int
+
+    @staticmethod
+    def validate_levels(min_level, max_level, amount_of_levels):
```

perhaps we should declare this outside of the dtos as a helper function?

---

we should raise invalid combinations e.g. max_level < min_level

---

```
if min is None or max is None:
    if amount_of_levels == 1:
         min, max = 1, 1
    if amount_of_levels >= 2:
          min, max = 1, amount_of_levels 
 if max < min or amount_of_levels < 1:
     raise ValueError
  ```